core/src/main/java/com/linecorp/armeria/server/http/dynamic/DynamicHttpServiceBuilder.java

```diff
@@ -0,0 +1,191 @@
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Sets;
+
+import com.linecorp.armeria.common.http.DefaultHttpResponse;
+import com.linecorp.armeria.common.http.HttpData;
+import com.linecorp.armeria.common.http.HttpHeaderNames;
+import com.linecorp.armeria.common.http.HttpHeaders;
+import com.linecorp.armeria.common.http.HttpMethod;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * Builds a new {@link DynamicHttpService}.
+ */
+public class DynamicHttpServiceBuilder {
+
+    private final List<DynamicHttpFunctionEntry> entries = new ArrayList<>();
+
+    /**
+     * Adds a new dynamic-mapped method, invoked with given (dynamically-bound) path.
+     */
+    public DynamicHttpServiceBuilder add(String path, DynamicHttpFunction function) {
+        return add(EnumSet.allOf(HttpMethod.class), path, function);
+    }
+
+    /**
+     * Adds a new dynamic-mapped method, invoked with given {@link HttpMethod} and (dynamically-bound) path.
+     */
+    public DynamicHttpServiceBuilder add(HttpMethod method, String path, DynamicHttpFunction function) {
+        return add(EnumSet.of(method), path, function);
+    }
+
+    /**
+     * Adds a new dynamic-mapped method, invoked with given {@link HttpMethod}s and (dynamically-bound) path.
+     */
+    public DynamicHttpServiceBuilder add(EnumSet<HttpMethod> methods, String path,
+                                         DynamicHttpFunction function) {
+        DynamicHttpFunctionEntry entry = new DynamicHttpFunctionEntry(methods, DynamicPath.of(path), function);
+        validate(entry);
+        entries.add(entry);
+        return this;
+    }
+
+    /**
+     * Add the {@link Path} annotated methods in given object.
+     *
+     * @see Path
+     * @see PathParam
+     */
+    public DynamicHttpServiceBuilder add(Object strategy) {
+        for (Method method : Methods.requestMappingMethods(strategy)) {
+            Path mapping = method.getAnnotation(Path.class);
+            EnumSet<HttpMethod> methods = Methods.httpMethods(method);
+            String mappedTo = mapping.value();
+            DynamicPath dynamicPath = DynamicPath.of(mappedTo);
+            int parameterCount = method.getParameterCount();
+            Class<?>[] parameterTypes = Methods.parameterTypes(method);
+            String[] pathVariables = Methods.pathVariables(method);
+            assert parameterCount == parameterTypes.length;
+            assert parameterCount == pathVariables.length;
+            Set<String> missingVariables = Sets.difference(Sets.newHashSet(pathVariables),
```

Comment:
Hmm, I see. I fix all the other Guava container methods. It seems like you prefer specifying the container type explicitly. (or, using container constructor explicitly.)